python/lsst/ts/externalscripts/base_take_twilight_flats.py

```diff
@@ -0,0 +1,457 @@
+
+        coords = coordinates.SkyCoord(
+            ra=_table["RAJ2000"],
+            dec=_table["DEJ2000"],
+            unit=(u.hourangle, u.deg),
+            frame=coordinates.ICRS,
+        )
+
+        arg = target.separation(coords).argmin()
+
+        return coords[arg]
+
+    def assert_sun_location(self):
+        """Confirm sun's elevation is safe for taking twilight flats."""
+        sun_coordinates = self.tcs.get_sun_azel()
+        where_sun = "setting" if (sun_coordinates[0] > 180) else "rising"
+        self.log.debug(
+            f" The azimuth of the {where_sun} Sun is {sun_coordinates[0]:.2f} deg \n"
+            f" The elevation of the Sun is {sun_coordinates[1]:.2f} deg"
+        )
+
+        if (sun_coordinates[1] < self.config.min_sun_elevation) or (
+            sun_coordinates[1] > self.config.max_sun_elevation
+        ):
+            raise AssertionError(
+                f"Sun elevation {sun_coordinates} is outside appropriate elvation limits. Aborting."
+            )
+
+    async def take_twilight_flats(self):
+
+        group_id = self.group_id if self.obs_id is None else self.obs_id
+
+        self.assert_sun_location()
+
+        target = self.get_target_radec()
+
+        # get an empty field
+        search_area_degrees = 10
+
+        empty_field_coords = self.get_empty_field(target, radius=search_area_degrees)
+        if empty_field_coords is not None:
+            self.log.info(
+                f"ICRS Empty field coordinates:\n"
+                f"  RA  = {empty_field_coords.ra.to_string(u.hour, sep=':')} ;"
+                f" DEC = {empty_field_coords.dec.to_string(u.degree, alwayssign=True, sep=':')}"
+            )
+
+            # Setup instrument filter and slew to desired field
+            await self.track_radec_and_setup_instrument(
+                empty_field_coords.ra, empty_field_coords.dec
+            )
+        else:
+            self.log.info("No empty field found. Continuing with default coords.")
+            await self.track_radec_and_setup_instrument(target.ra, target.dec)
+
+        # Take one 1s flat to calibrate the exposure time
+        self.log.info("Taking 1s flat to calibrate exposure time.")
+        exp_time = 1
+        flat_image = await self.camera.take_acq(
+            exptime=exp_time,
+            n=1,
+            group_id=group_id,
+            program=self.program,
+            reason=self.reason,
+        )
+
+        self.latest_exposure_id = int(flat_image[0])
+
+        for i in range(self.config.n_flat):
+
+            sky_counts = await self.get_sky_counts()
+            self.log.info(
+                f"Flat just taken with exposure time {exp_time} had counts of {sky_counts}"
+            )
+
+            exp_time = self.get_new_exptime(sky_counts, exp_time)
+
+            if exp_time > self.config.max_exp_time:
+                self.log.warning(
+                    f"Calculated exposure time {exp_time} above max exposure time. Aborting."
```

Comment:
Please, log the value of `max_exp_time`, e.g. 

```
f"Calculated exposure time {exp_time} above max exposure time {self.config.max_exp_time}. Aborting."
```

Also, consider replacing "Aborting" with "Stopping" or "Interrupting". Aborting usually means something very strong but here you are only breaking the loop.